data/routers/action-router.js

```diff
@@ -8,108 +8,128 @@ router.get('/', async (req, res) => {
+    res.status(500).json({
+      error: `There was an error while checking the project ID. ${err}`
     });
   }
-  // Then check if the action submission requirements are met:
-  if (!req.body.description || !req.body.notes) {
+  if (!description || !notes) {
     res.status(400).json({
       error: 'Please provide both a description and notes for the action.'
     });
-  } else if (req.body.description.length > 128) {
+  } else if (description.length > 128) {
     res.status(400).json({
       error:
-        'Your action description is limited to 128 characters. Please use the notes field for additional information.'
-    });
-  }
-  // Then submit the new action:
-  try {
-    const addedAction = await actionModel.insert({
-      project_id: req.params.id,
-      description: req.body.description,
-      notes: req.body.notes
-    });
-    res.status(201).json(addedAction);
-  } catch (err) {
-    res.status(500).json({
-      error: 'There was an error while saving the action to the database.'
+        'The action description field is limited to 128 characters. Please use the action notes field to add additional information.'
     });
+  } else {
+    try {
+      const addedAction = await actionModel.insert(req.body);
+      res.status(201).json(addedAction);
+    } catch (err) {
+      res.status(500).json({
+        error: `There was an error while saving the action. ${err}`
+      });
+    }
   }
 });
 
 router.put('/:id', async (req, res) => {
-  if (!req.params.id || !req.body.changes) {
+  const { project_id, description, notes, completed } = req.body;
+  // Make the first submission check before making a server call:
+  if (!project_id && !description && !notes && !completed) {
+    res.status(400).json({
+      error: 'Please provide the action changes you intend to make.'
+    });
+    // Make the second submission check before making a server call:
+  } else if (description && description.length > 128) {
     res.status(400).json({
       error:
-        'Please provide the ID of the action you intend to update as well as your intended changes.'
+        'The action description field is limited to 128 characters. Please use the action notes field to add additional information.'
     });
-  } else if (req.body.changes.description) {
-    if (req.body.changes.description.length > 128) {
-      res.status(400).json({
-        error:
-          'Your action description is limited to 128 characters. Please use the notes field for additional information.'
-      });
+  } else {
+    // If the first two submission checks pass, make the next submission check (this one requires a server call):
+    if (project_id) {
+      try {
+        const project = await projectModel.get(project_id);
+        if (!project) {
+          // The cleanest way I could think of to cover this check was to add a return here...:
+          return res.status(400).json({
```

Comment:
yes, if you have an `if` statement without a paired `else` block, you need to `return` the response here